bbc/simorgh
# Canonical referral link

Resolves #1972 

**Overall change:** Passed previous pathname value to the ATIAnalytics module.

**Code changes:**

- Updated unit tests
- added effect for saving value in the state during unmounting
- passed value to request context
- use value in ati analytics container
- add logic to distinguish different user journeys
- add check to clear previous path on back clicks ( see [previous review](https://github.com/bbc/simorgh/pull/2145))
---
**To Test**
- download the [`tag inspector` add-on](https://chrome.google.com/webstore/detail/at-internet-tag-inspector/epdfbeoiphkaeapcohmilhmpdeilgnok?hl=en) in chrome (or check the ATI outbound request in the network tab)
- visit http://localhost:7080/news/articles/c0g992jmmkko and check the ATI beacon's referrer (field `x6` in the outbound request) is not set
- click the link in the body of the article `placed on the tomb of the unknown warrior` which links through to http://localhost:7080/news/articles/c6v11qzyv8po. Check the referrer (`x6`) of the new page is set to the previous page `http://localhost:7080/news/articles/c0g992jmmkko`
- Finally click the link `which were later donated to local hospices` which links through to http://localhost:7080/news/articles/c0g992jmmkko. Again check the new beacon's referrer is set to the previous page this time `c0g992jmmkko`

---
